diff --git a/APLS Elan File Checker/app.R b/APLS Elan File Checker/app.R
index c872c9e..067f862 100644
--- a/APLS Elan File Checker/app.R	
+++ b/APLS Elan File Checker/app.R	
@@ -520,21 +537,21 @@ findOverlaps <- function(timesEAF, eafName) {
 ##  annotation times (one file's worth) and a single EAF name (meant to be used
 ##  with output of getTimes() and imap()), and rotates through tiers, fixing
 ##  overlaps, until it reaches a stable state
-fixOverlaps <- function(timesEAF, eafName, eaflist) {
-  
[...]

+  ##Ensure tierOrder matches overlapsInit names
+  if (!identical(sort(names(overlapsInit)),
+                 sort(tierOrder))) {
+    stop("names(overlapsInit): ", names(overlapsInit),
+         "\ntierOrder: ", tierOrder,
+         "\nnonSpkrTiers: ", nonSpkrTiers)
+  }
   
   ##Initialize looping variables
   overlapsPre <- NULL
@@ -554,15 +571,34 @@ fixOverlaps <- function(timesEAF, eafName, eaflist) {
     ##Old post is new pre
     overlapsPre <- overlapsPost
     
-    ##Go through another fixing round
-    overlapsPost <- 
-      overlapsPre %>%
-      map(fixOverlapsTier, eaflist, eafName) %>% 
-      map(filter, !Resolved)
+    ##Fix each tier in turn
+    for (tier in rev(tierOrder)) {
+      ##Re-assess overlaps now that eaflist has been modified
+      overlapsCurr <- 
+        eaflist %>%
+        pluck(eafName) %>% 
+        getTimes(eafName=eafName, tiers=tierOrder) %>% 
+        findOverlaps(eafName)
+      
+      ##Hide overlaps in all other tiers from fixOverlapsTier()
+      for (otherSpkr in setdiff(names(overlapsPre), tier)) {
+        overlapsCurr <- overlapsCurr %>% 
+          modify_in(list(otherSpkr), ~ .x %>% filter(is.na(ANNOTATION_ID)))
+      }
+      
+      ##Fix overlaps for *other* tiers, keeping current tier
+      overlapsPost[[tier]] <-
+        overlapsCurr %>% 
+        map(fixOverlapsTier, eaflist, eafName) %>% 
+        pluck(tier) %>% 
+        filter(!Resolved)
+    }
     
   }
   
   ##Return overlapsPost
@@ -587,22 +623,41 @@ formatTimes <- function(time, type=c("S","HMS")[2]) {
 ##  nicely for printing
 ##x should be eaflist(), df should be tierInfo()
 overlapsIssues <- function(x, df) {
+  ##Get tier order by file
+  tierOrder <- df %>% 
+    mutate(OverlapTier = case_when(
+      TIER_ID=="Redaction" ~ "Redaction",
+      TIER_ID==SpkrCode ~ "Main speaker",
+      Neighborhood=="HD" & TIER_ID=="Trista Pennington" ~ "Interviewer",
+      Neighborhood!="HD" & TIER_ID=="Barbara Johnstone" ~ "Interviewer",
+      startsWith(TIER_ID, "Interviewer") ~ "Interviewer",
+      SpkrTier ~ "Bystander",
+      TRUE ~ NA_character_
+    )) %>% 
+    filter(!is.na(OverlapTier)) %>% 
+    arrange(desc(OverlapTier)) %>% 
+    select(File, TIER_ID) %>% 
+    nest(tierOrder = TIER_ID) %>% 
+    pull(tierOrder, File) %>% 
+    map("TIER_ID")
+  
   ##Get timing data

